src/async_session.rs

```diff
@@ -57,6 +57,33 @@ impl AsyncSession {
             e => panic!("WaitForMultipleObjects returned unexpected value {:?}", e),
         }
     }
+
+    pub async fn recv(&self, buf: &mut [u8]) -> std::io::Result<usize> {
+        loop {
+            match self.session.try_receive() {
+                Ok(Some(packet)) => {
+                    // it seems to be standard practice to truncate rather than return an error,
+                    // in the context of TUN/TAP the packet lengths can be read anyway
+                    let size = packet.bytes.len().min(buf.len());
```

Comment:
I still haven't figured out how to handle errors here. Can you suggest a better way?